packages/plugins/src/hardhat/simulate.ts

```diff
@@ -237,11 +249,93 @@ export const simulateDeploymentSubtask = async (
     throw new Error(`Unknown execution mode.`)
   }
 
-  const deploymentData = makeDeploymentData(parsedConfigArray)
+  const deploymentData = makeDeploymentData(compilerConfigArray)
   const merkleTree = makeSphinxMerkleTree(deploymentData)
 
-  const { receipts, batches, finalStatus, failureAction } =
-    await runEntireDeploymentProcess(parsedConfig, merkleTree, provider, signer)
+  await fundAccountMaxBalance(signer.address, provider)
+  await setManagedServiceRelayer(signer.address, provider)
```

Comment:
Seems like `setManagedServiceRelayer` is redundant since we already call it in `injectRoles`?